keybase/source.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package keybase
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/keybase/go-logging"
+	"github.com/keybase/go-updater"
+	"github.com/keybase/go-updater/util"
+)
+
+type responseStatus struct {
+	Code   int               `json:"code"`
+	Name   string            `json:"name"`
+	Desc   string            `json:"desc"`
+	Fields map[string]string `json:"fields"`
+}
+
+type updateResponse struct {
+	Status responseStatus `json:"status"`
+	Update updater.Update `json:"update"`
+}
+
+// UpdateSource finds releases/updates on keybase.io
+type UpdateSource struct {
+	log      logging.Logger
+	endpoint string
+}
+
+// NewUpdateSource contructs an update source for keybase.io
+func NewUpdateSource(log logging.Logger) UpdateSource {
+	return newUpdateSource("https://keybase.io/_/api/1.0/pkg/update.json", log)
```

Comment:
That requires adding our cert... created story for that here: https://keybase.atlassian.net/browse/DESKTOP-925